sw/airborne/subsystems/ins/ins_int.c

```diff
@@ -510,37 +508,43 @@ static void vel_est_cb(uint8_t sender_id __attribute__((unused)),
                        float x, float y, float z,
                        float noise __attribute__((unused)))
 {
-
+#if USE_HFF
   struct FloatVect3 vel_body = {x, y, z};
-  static uint32_t last_stamp = 0;
-  float dt = 0;
 
   /* rotate velocity estimate to nav/ltp frame */
   struct FloatQuat q_b2n = *stateGetNedToBodyQuat_f();
   QUAT_INVERT(q_b2n, q_b2n);
   struct FloatVect3 vel_ned;
   float_quat_vmult(&vel_ned, &q_b2n, &vel_body);
 
+  struct FloatVect2 vel = {vel_ned.x, vel_ned.y};
+  struct FloatVect2 Rvel = {noise, noise};
+
+  b2_hff_update_vel(vel,  Rvel);
+  ins_update_from_hff();
+#else
+  /* timestamp in usec when last callback was received */
+  static uint32_t last_stamp = 0;
+  float dt = 0;
+
   if (last_stamp > 0) {
     dt = (float)(stamp - last_stamp) * 1e-6;
   }
-
   last_stamp = stamp;
 
-#if USE_HFF
-  (void)dt; //dt is unused variable in this define
+  struct Int32Vect3 vel_body = {SPEED_BFP_OF_REAL(x), SPEED_BFP_OF_REAL(y), SPEED_BFP_OF_REAL(z)};
 
-  struct FloatVect2 vel = {vel_ned.x, vel_ned.y};
-  struct FloatVect2 Rvel = {noise, noise};
+  /* rotate velocity estimate to nav/ltp frame */
+  struct Int32Quat q_b2n = *stateGetNedToBodyQuat_i();
+  QUAT_INVERT(q_b2n, q_b2n);
+  struct Int32Vect3 vel_ned;
+  int32_quat_vmult(&vel_ned, &q_b2n, &vel_body);
 
-  b2_hff_update_vel(vel,  Rvel);
-  ins_update_from_hff();
-#else
-  ins_int.ltp_speed.x = SPEED_BFP_OF_REAL(vel_ned.x);
-  ins_int.ltp_speed.y = SPEED_BFP_OF_REAL(vel_ned.y);
+  ins_int.ltp_speed.x = vel_ned.x;
```

Comment:
ah right, then I would suggest renaming it `vel_ned_i` or so to make it more readable...